cmd/topicmappr/commands/reassignments.go

```diff
@@ -288,6 +288,55 @@ func computeReassignmentBundles(
 	return results
 }
 
+/**
+getPartitionMapChunk Breaks a reassignment into a series of sequential, smaller reassignments.
+For large reassignments that may take a while, or risky operations that may require downtime in between, a chunked reassignment can be used.
+This will generate a series of partition maps that will converge on the desired state. To minimize data transfer,
+partitions are only moved to replicas in the desired final state.
+
+The original design was intended for downscaling operations, to remove partitions from one (or three) brokers at a time,
+without overwhelming whatever brokers are remaining in the cluster.
+*/
+func getPartitionMapChunk(finalMap *kafkazk.PartitionMap, initialMap *kafkazk.PartitionMap, brokerIds kafkazk.BrokerList, chunkStepSize int) []*kafkazk.PartitionMap {
+	var intermediateMap = initialMap.Copy()
+	var out []*kafkazk.PartitionMap
+	brokerIds.SortByIDDesc()
+
+	for i := 0; i < len(brokerIds); i += chunkStepSize {
```

Comment:
In case we care to (I haven't looked thoroughly at how the indexes are referenced), we could combine the outer loop and chunk range selection in a single, bounds-safe way: https://go.dev/play/p/-N3C9RFhyh8

Reply:
That is nice, but I can't figure out a nice way to make the `contains` check later work with that prettily